### PR TITLE
PERF: make Asset comparison faster

### PR DESCRIPTION
**Pull request by llllllllll**

### CODE REVIEW

**Comment by richafrank** [comment]
LGTM :+1:

File: zipline/assets/_assets.pyx [hunk 1]
```diff
@@ -75,58 +87,37 @@ cdef class Asset:
...
-        if isinstance(x, Asset):
-            x_as_int = x.sid
-        elif isinstance(x, int):
-            # ints are Integral, but much faster to special case
-            x_as_int = x
-        elif isinstance(x, (long, np.int64, np.int32, Integral)):
-            x_as_int = int(x)
-        else:
+        try:
+            x_as_int = PyNumber_Index(x)
```
Comment (richafrank) [inline zipline/assets/_assets.pyx]: Are we ok with:
```python
>>> 1 == 9999999999L
False
>>> Asset(1) == 9999999999L
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "zipline/assets/_assets.pyx", line 99, in zipline.assets._assets.Asset.__richcmp__ (zipline/assets/_assets.c:2290)
    y_as_int = y.sid
OverflowError: value too large to convert to int
```

Comment (richafrank) [inline zipline/assets/_assets.pyx]: FWIW That doesn't seem to be new to this PR:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "zipline/assets/_assets.pyx", line 104, in zipline.assets._assets.Asset.__richcmp__ (zipline/assets/_assets.c:2348)
    y_as_int = int(y)
OverflowError: value too large to convert to int
```

Comment (llllllllll) [inline zipline/assets/_assets.pyx]: That seems reasonable. The alternative would be to make this a bigger type like uint64 but idk if that is worth it and only moves the boundry up. The alternative is to catch this and return `NotImplemented` which will propagate as a type error. I think that error is less informative.

**Pull request merged**